### PR TITLE
Feature/remove api error

### PR DESCRIPTION
**Pull request by le-binh**
Description
--
- Remove `APIError` and  use `TweepyError` instead
- Add `StatusDuplicateError` which indicate that is a duplicate status

### CODE REVIEW

File: responsebot/responsebot_stream.py [hunk 1]
```diff
@@ -50,7 +49,7 @@ def start(self, retry_limit=None):
             except TweepError as e:
                 if 'Stream object already connected!' in e.reason or \
                         'Wrong number of locations points' in e.reason:
-                    raise APIError(e.reason)
+                    raise
```
Comment (nhanbt) [inline responsebot/responsebot_stream.py]: This code will silently continue without logging any error message at all if the `TweepError` isn't one of the 2 errors being checked by the `if`, won't it?

To be fair this problem is present in the old code too. Even if continuing is really the intended behavior, I guess it's better to at least log the error out. Maybe something like:

```python
else
    logging.error('Skipping API error: %s' % str(e))
```

File: responsebot/handlers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from .base import BaseTweetHandler
+from .event import BaseEventHandler
+
+
+registered_handlers = set([])
```
Comment (nhanbt) [inline responsebot/handlers/__init__.py]: Nitpicking here but isn't `set()` enough?

Comment (steadykhoi) [reply responsebot/handlers/__init__.py]: Yes `set()` is enough.

File: responsebot/responsebot_stream.py (hunk 1 above)
Comment (steadykhoi) [reply responsebot/responsebot_stream.py]: I'm sorry, we should remove this except clause entirely. We only want to raise the `TweepError` whatever it's reason was.

**Comment by nhanbt** [comment]
LGTM. Thanks guys!

**Pull request merged**